internal/memsync/shell_test.go

```
@@ -16,18 +20,131 @@ import (
+// swallows `rmdir`'s stderr (`2>/dev/null; exit 0`), so an implementation that
+// takes the flags and prunes nothing arrives as a surviving directory rather
+// than as an error — precisely how #554 mis-reported itself. One residue is
+// left standing: a `sort` that accepts `-z` and does not sort passes both the
+// probe and the tests, which compare the listing as a map and never assert the
+// byte order the command promises.
+func hashTreePrereqs(t *testing.T) {
+	t.Helper()
+	dir := t.TempDir()
+	requirePrereq(t, "bash", exec.Command("bash", "-c", "exit 0").Run())
+	requirePrereq(t, "find -print0", exec.Command("find", dir, "-type", "f", "-print0").Run())
+	requirePrereq(t, "sort -z", exec.Command("sort", "-z", "/dev/null").Run())
```

golangci-lint (2.13.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)